quantum/quantum.h

```diff
@@ -135,70 +133,60 @@ extern uint32_t default_layer_state;
     #include "hd44780.h"
 #endif
 
-//Function substitutions to ease GPIO manipulation
-#ifdef __AVR__
-    #define PIN_ADDRESS(p, offset) _SFR_IO8(ADDRESS_BASE + (p >> PORT_SHIFTER) + offset)
-
-    #define pin_t uint8_t
-    #define setPinInput(pin) PIN_ADDRESS(pin, 1) &= ~ _BV(pin & 0xF)
-    #define setPinInputHigh(pin) ({\
-            PIN_ADDRESS(pin, 1) &= ~ _BV(pin & 0xF);\
-            PIN_ADDRESS(pin, 2) |=   _BV(pin & 0xF);\
-            })
-    #define setPinInputLow(pin) _Static_assert(0, "AVR Processors cannot impliment an input as pull low")
-    #define setPinOutput(pin) PIN_ADDRESS(pin, 1) |= _BV(pin & 0xF)
-
-    #define writePinHigh(pin) PIN_ADDRESS(pin, 2) |=  _BV(pin & 0xF)
-    #define writePinLow(pin) PIN_ADDRESS(pin, 2) &= ~_BV(pin & 0xF)
-    static inline void writePin(pin_t pin, uint8_t level){
```

Comment:
This inline function has been replaced with a macro. The only difference in behavior is with single vs multiple evaluation of arguments, but a quick grep reveals that `writePin` isn't used anywhere in a way where this could matter (e.g. `writePin(p++)`).

<pre><details>
<summary>rg 'writePin' -g '!docs'</summary>
quantum/matrix.c
264:    writePinLow(row_pins[row]);
331:    writePinLow(col_pins[col]);

quantum/quantum.h
147:    #define writePinHigh(pin)       (PIN_ADDRESS(pin, 2) |=  _BV((pin) & 0xF))
148:    #define writePinLow(pin)        (PIN_ADDRESS(pin, 2) &= ~_BV((pin) & 0xF))
149:    #define writePin(pin, level)    ((level) ? writePinHigh(pin) : writePinLow(pin))
160:    #define writePinHigh(pin)       palSetLine(pin)
161:    #define writePinLow(pin)        palClearLine(pin)
162:    #define writePin(pin, level)    ((level) ? writePinHigh(pin) : writePinLow(pin))

keyboards/e6_rgb/e6_rgb.c
21:  writePinHigh(D5);

keyboards/e7v1/e7v1.c
18:    writePinHigh(F0);
20:    writePinLow(F0);

keyboards/proton_c/proton_c.c
22:  writePinHigh(C13);

keyboards/touchpad/matrix.c
135:  writePinHigh(B7);
234:        case 0: writePin(D6, matrix_is_on(r, c)); break;
235:        case 1: writePin(B4, matrix_is_on(r, c)); break;
236:        case 2: writePin(B5, matrix_is_on(r, c)); break;
237:        case 3: writePin(B6, matrix_is_on(r, c)); break;
238:        case 4: writePin(C6, matrix_is_on(r, c)); break;
239:        case 5: writePin(C7, matrix_is_on(r, c)); break;
243:        case 0: writePin(F5, !matrix_is_on(r, c)); break;
244:        case 1: writePin(F4, !matrix_is_on(r, c)); break;
245:        case 2: writePin(F1, !matrix_is_on(r, c)); break;
246:        case 3: writePin(F0, !matrix_is_on(r, c)); break;
247:        case 4: writePin(F6, !matrix_is_on(r, c)); break;
248:        case 5: writePin(F7, !matrix_is_on(r, c)); break;
254:    writePinHigh(E6);
255:    writePinHigh(D7);
260:    writePinLow(D7);
261:    writePinLow(E6);

quantum/split_common/matrix.c
146:  writePinLow(row_pins[row]);
191:  writePinLow(col_pins[col]);

keyboards/westfoxtrot/aanzee/aanzee.c
40:  writePinLow(B2);
46:  writePinHigh(B2);

layouts/community/ortho_4x12/drashna/keymap.c
227:    writePinHigh(D5);
230:    writePinHigh(B0);

layouts/community/ortho_5x12/drashna/keymap.c
151:    writePinHigh(D5);
154:    writePinHigh(B0);

keyboards/crkbd/keymaps/drashna/keymap.c
98:    writePinHigh(D5);
101:    writePinHigh(B0);

keyboards/iris/keymaps/drashna/keymap.c
94:    writePinHigh(D5);
97:    writePinHigh(B0);

keyboards/orthodox/keymaps/drashna/keymap.c
102:    writePinHigh(D5);
105:    writePinHigh(B0);

keyboards/viterbi/keymaps/drashna/keymap.c
94:    writePinHigh(D5);
97:    writePinHigh(B0);
</details></pre>